dll/win32/shell32/wine/shellord.c

```diff
@@ -403,36 +418,52 @@ int ShellMessageBoxA(
 	UINT uType,
 	...)
 {
-	char	szText[100],szTitle[100];
-	LPCSTR  pszText = szText, pszTitle = szTitle;
-	LPSTR   pszTemp;
-	__ms_va_list args;
-	int	ret;
+    char    *szText = NULL,szTitle[100];
+    LPCSTR  pszText = szText, pszTitle = szTitle;
+    LPSTR   pszTemp;
+    __ms_va_list args;
+    int    ret;
 
-	__ms_va_start(args, uType);
-	/* wvsprintfA(buf,fmt, args); */
+    __ms_va_start(args, uType);
+    /* wvsprintfA(buf,fmt, args); */
 
-	TRACE("(%p,%p,%p,%p,%08x)\n",
-	    hInstance,hWnd,lpText,lpCaption,uType);
+    TRACE("(%p,%p,%p,%p,%08x)\n",
+        hInstance,hWnd,lpText,lpCaption,uType);
 
-	if (IS_INTRESOURCE(lpCaption))
-	  LoadStringA(hInstance, LOWORD(lpCaption), szTitle, sizeof(szTitle));
-	else
-	  pszTitle = lpCaption;
+    if (IS_INTRESOURCE(lpCaption))
+      LoadStringA(hInstance, LOWORD(lpCaption), szTitle, _countof(szTitle));
+    else
+      pszTitle = lpCaption;
 
-	if (IS_INTRESOURCE(lpText))
-	  LoadStringA(hInstance, LOWORD(lpText), szText, sizeof(szText));
-	else
-	  pszText = lpText;
+    if (IS_INTRESOURCE(lpText))
+    {
+        const char *ptr;
+        UINT len = LoadStringA(hInstance, LOWORD(lpText), (LPSTR)&ptr, 0);
```

Comment:
Didn't know the A version could do this...

Reply:
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadstringa ?

Reply:
Seems that MSDN is not always right :)
The A version does not do this (how could it, the string is not stored as Ansi version in the resources?),
only the W version is doing this.

Reply:
Oups :-)